### PR TITLE
Update Doros for new entries

### PR DESCRIPTION
**Pull request by JoschD**
The Doros team added more entries on the root-level of the HDF5 files.
Before, there were only BPMs and `METADATA` and the BPMs were identified by not being `METADATA`.
As now there is also `TIMESTAMPS_INDEX` and `TIMESTAMPS_TABLE` (I told Manuel this maybe should go into `METADATA`, as `METADATA` itself seems to be empty anyway... but so far they have not moved it).

I considered going by name (e.g. find `_DOROS` in the name) but then we are limited to which machine data we can write into this format.
I now identify BPMs as entries that have a subgroup `nbOrbitSamplesRead` which should be pretty robust.

Added a test with a reduced version of the new data (script to generate the data is also attached).

### CODE REVIEW

File: turn_by_turn/doros.py [hunk 1]
```diff
@@ -85,7 +86,8 @@ def read_tbt(file_path: str|Path, bunch_id: int = DEFAULT_BUNCH_ID) -> TbtData:
     LOGGER.debug(f"Reading DOROS file at path: '{file_path.absolute()}'")
     with h5py.File(file_path, "r") as hdf_file:
         # use "/" to keep track of bpm order, see https://github.com/h5py/h5py/issues/1471
-        bpm_names = [name for name in hdf_file["/"].keys() if name != METADATA]
+        bpm_names = [name for name in hdf_file["/"].keys() if N_ORBIT_SAMPLES in hdf_file[f"/{name}"].keys()]
+        LOGGER.debug(f"Found BPMs in DOROS-type file: {bpm_names}")
```
Comment (fsoubelet) [inline turn_by_turn/doros.py:90]: All fine by me.

File: tests/inputs/doros_shrink_data.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Quick script to remove some data from the DOROS files, to make them smaller for the tests.
+"""
+from pathlib import Path
+import shutil
+import subprocess
+import h5py
+
+from turn_by_turn.doros import DEFAULT_OSCILLATION_DATA, N_ORBIT_SAMPLES, N_OSCILLATION_SAMPLES, OSCILLATIONS, POSITIONS
+
+file_in = "/afs/cern.ch/work/j/jdilly/dorostest/DOROS-2024-09-29_01_37_13_522358-NO_USER.h5"
```
Comment (fsoubelet) [inline tests/inputs/doros_shrink_data.py:11]: Don't like a hardcoded path here

File: tests/inputs/doros_shrink_data.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+N_BPMS= 3
+NTURNS = 50000
+
+# Copy file to temporary location and "delete" unneccessary data (only removes references to data)
+file_temp = file_out + ".tmp"
+shutil.copyfile(file_in, file_temp)
+with h5py.File(file_temp, "r+", track_order=True) as hdf_file:
+    bpms = [name for name in hdf_file["/"].keys() if N_ORBIT_SAMPLES in hdf_file[f"/{name}"].keys()]
+
+    for bpm in bpms[:N_BPMS]:
```
Comment (fsoubelet) [inline tests/inputs/doros_shrink_data.py:24]: Done so we just keep 3 BPMs worth of data, right? Keeping size reasonable and all

File: tests/inputs/doros_shrink_data.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        del hdf_file[bpm][OSCILLATIONS["Y"]]
+        hdf_file[bpm].create_dataset(N_OSCILLATION_SAMPLES, data=0)
+        hdf_file[bpm].create_dataset(OSCILLATIONS["X"], data=[DEFAULT_OSCILLATION_DATA])
+        hdf_file[bpm].create_dataset(OSCILLATIONS["Y"], data=[DEFAULT_OSCILLATION_DATA])
+    
+    for bpm in bpms[N_BPMS:]:
+        del hdf_file[bpm]
+
+# Repack file (which actually removes the data in the file)
+subprocess.run(["h5repack", file_temp, file_out])
```
Comment (fsoubelet) [inline tests/inputs/doros_shrink_data.py:46]: Maybe at least add a comment at the top of the file that this package is needed? I'm not sure what error is raised by `subprocess` when that fails, might not be super helpful.

**Review by fsoubelet — COMMENTED** [review]
Regarding the `h5repack` thing: how about we just create a new HDF5 file from scratch, and write into it the relevant parts? This way without copying first we don't have to worry about data deletion in HDF5. What do you think?

File: tests/inputs/doros_shrink_data.py (hunk 3 above)
Comment (JoschD) [reply tests/inputs/doros_shrink_data.py:24]: yes, exactly. And I choose also 3 bpms as this is checked in the test (as our fake-data also has 3 BPMs) so I didn't need to rewrite the test for this case and coudl re-use for both files

File: tests/inputs/doros_shrink_data.py (hunk 2 above)
Comment (JoschD) [reply tests/inputs/doros_shrink_data.py:11]: I know. But it's just a script. people will have to hardcode their own path to run it anyway

**Comment by JoschD** [comment]
> Regarding the h5repack thing: how about we just create a new HDF5 file from scratch, and write into it the relevant parts? This way without copying first we don't have to worry about data deletion in HDF5. What do you think?

I was thinking about it, and even trying it ... but then you kind of need to know what the other data is, that is in the file and how many "levels" deep it is in the hdf5, as you cannot simply copy a whole "tree" into another file. And if we build up a fake-file from scratch, we might miss something that they (the DOROS) people put in that make the reader fail (as was the case with the new entries leading to this PR). 
This script is only used in case we ever need to update the test-file anyway. I don't care if it crashes or is not the most optimal. I wanted a quick way to reduce the 20-30MB DOROS files while keeping their internal structure as intact as possible.

File: tests/inputs/doros_shrink_data.py (hunk 4 above)
Comment (fsoubelet) [reply tests/inputs/doros_shrink_data.py:46]: Seems error from subprocess is enough for the user.

**Review by fsoubelet — APPROVED** [review]
In this case I'm all ok with this PR, thanks!

**Pull request merged**